crates/holochain_keystore/src/lib.rs

```diff
@@ -44,27 +19,63 @@ pub mod lair_keystore;
 
 pub mod paths;
 
+#[cfg(feature = "test_utils")]
 mod test_keystore;
+#[cfg(feature = "test_utils")]
 pub use test_keystore::*;
 
 #[cfg(feature = "test_utils")]
 pub mod crude_mock_keystore;
 
 /// Construct a simple in-memory in-process keystore.
+///
+/// # Examples
+///
+/// ```
+/// use holo_hash::AgentPubKey;
+/// use holochain_keystore::*;
+/// use holochain_serialized_bytes::prelude::*;
+///
+/// #[tokio::main(flavor = "multi_thread")]
+/// async fn main() {
+///     tokio::task::spawn(async move {
+///         let keystore = spawn_mem_keystore().await.unwrap();
+///         let agent_pubkey = AgentPubKey::new_random(&keystore).await.unwrap();
+///
+///         #[derive(Debug, serde::Serialize, serde::Deserialize, SerializedBytes)]
+///         struct MyData(Vec<u8>);
+///
+///         let my_data_1 = MyData(b"signature test data 1".to_vec());
+///
+///         let signature = agent_pubkey.sign(&keystore, &my_data_1).await.unwrap();
+///
+///         assert!(agent_pubkey.verify_signature(&signature, &my_data_1).await.unwrap());
+///     }).await.unwrap();
+/// }
+/// ```
+#[cfg(feature = "test_utils")]
 pub async fn spawn_mem_keystore() -> LairResult<MetaLairClient> {
-    use kitsune_p2p_types::dependencies::lair_keystore_api;
-    use lair_keystore_api::prelude::*;
-    use std::sync::Arc;
-
     // in-memory secure random passphrase
     let passphrase = sodoken::BufWrite::new_mem_locked(32)?;
     sodoken::random::bytes_buf(passphrase.clone()).await?;
 
+    spawn_mem_keystore_with_passphrase(passphrase.to_read()).await
+}
+
+/// Construct a simple in-memory in-process keystore with the given passphrase.
+#[cfg(feature = "test_utils")]
+pub async fn spawn_mem_keystore_with_passphrase(
```

Comment:
This is what I came to change, there was no way to spawn a test keystore with a known password